### PR TITLE
Windows support for the --nightly | -n flag to mach run commands.

### PR DESCRIPTION
**Pull request by o0Ignition0o**
Add windows support to the -n flag.
---
Followup to #19947 , this PR will add windows support to the -n flag.
 This is part of step two for #19505 

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

The feature does not work just yet, I'm able to download and extract the archive, but it's not running the executable yet.
@tigercosmos might be a good reviewer on this one :) 

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because they're part of a ./mach command.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20122)

### CODE REVIEW

**Comment by o0Ignition0o** [comment]
Yeah definitely ! :)

File: python/servo/command_base.py [hunk 1]
```diff
@@ -457,11 +459,21 @@ def get_nightly_binary_path(self, nightly_date):
...
         else:
-            print("Extracting to {}...".format(destination_folder))
-            with tarfile.open(os.path.join(nightlies_folder, destination_file), "r") as tar:
-                tar.extractall(destination_folder)
-
-        return path.join(destination_folder, "servo", "servo")
+            print("Extracting to {} ...".format(destination_folder))
+            if is_windows():
+                command = 'msiexec /a {} /qn TARGETDIR={}' \
+                    .format(os.path.join(nightlies_folder, destination_file), destination_folder)
```
Comment (jdm) [inline python/servo/command_base.py]: Let's use this formatting instead:
```python
command = 'msiexec /a {} /qn TARGETDIR={}'.format(
    path.join(...), destination_folder)
```

File: python/servo/command_base.py [hunk 2]
```diff
@@ -457,11 +459,21 @@ def get_nightly_binary_path(self, nightly_date):
...
+                if subprocess.call(command, stdout=PIPE, stderr=PIPE) != 0:
+                    print("Could not extract the nightly executable from the msi package.")
+                    sys.exit(1)
+            else:
+                with tarfile.open(os.path.join(nightlies_folder, destination_file), "r") as tar:
+                    tar.extractall(destination_folder)
+        bin_folder = path.join(destination_folder, "servo")
+        if is_windows():
+            bin_folder = path.join(destination_folder, "PFiles", "Mozilla research", "Servo Tech Demo")
+        print(path.join(bin_folder, "servo{}".format(BIN_SUFFIX)))
```
Comment (jdm) [inline python/servo/command_base.py]: Do we still need this?

**Review by jdm — CHANGES_REQUESTED** [review]
This looks like a good improvement on the original PR. Thanks!

File: python/servo/command_base.py (hunk 2 above)
Comment (o0Ignition0o) [reply python/servo/command_base.py]: I think I could extract the .msi archive to %temp% and just keep the .exe file instead :)

Comment (jdm) [reply python/servo/command_base.py]: I was asking about the `print`, but it's worth a try!

Comment (o0Ignition0o) [reply python/servo/command_base.py]: Oh I see, no it's debug code I use (I seem to get the right path but when I try to ./mach run with a path to the executable it just won't ^^
I guess as soon as I figure it out I'll remove the WIP tag on the PR (Probably this weekend :))

**Comment by o0Ignition0o** [comment]
This should be reviewable now.
Please not I have _not_ been able to run the executable on my windows virtual machine (but I still couldn't do so when downloading and running it manually). So extra caution / trying to run it on windows would be more than appreciated ^^'

**Comment by jdm** [comment]
Is there a difference if you run a windows nightly since March 1? There was a two month period in which we broke nightlies on windows at startup.

**Comment by o0Ignition0o** [comment]
I got the same result as I ran 
> PS C:\Users\jerem\Documents\Mozilla\servo> .\mach.bat run -n 2018-03-08 https://mozilla.org 

The window seems to open and closes instantly.
The same behaviour happens when I run the Windows Build ZIP from https://download.servo.org/ so I'm pretty convinced it has to do with my setup

**Pull request merged**